submarine-sdk/pysubmarine/tests/store/model_registry/test_sqlalchemy_store.py

```diff
@@ -0,0 +1,731 @@
+            user_id="test",
+            experiment_id="application_1234",
+            tags=tags,
+        )
+        self.assertEqual(mv.creation_time, fake_datetime)
+        self.assertEqual(mv.last_updated_time, fake_datetime)
+        mvd = self.store.get_model_version(mv.name, mv.version)
+        self.assertEqual(mvd.name, name)
+        self.assertEqual(mvd.user_id, "test")
+        self.assertEqual(mvd.experiment_id, "application_1234")
+        self.assertEqual(mvd.current_stage, STAGE_NONE)
+        self.assertEqual(mvd.creation_time, fake_datetime)
+        self.assertEqual(mvd.last_updated_time, fake_datetime)
+        self.assertEqual(mvd.source, "path/to/source")
+        self.assertEqual(mvd.dataset, None)
+        self.assertEqual(mvd.description, None)
+        self.assertEqual(mvd.tags, tags)
+
+    def _compare_model_versions(self, results: List[ModelVersion], mvs: List[ModelVersion]) -> None:
+        result_versions = set([result.version for result in results])
+        model_versions = set([mv.version for mv in mvs])
+
+        self.assertEqual(result_versions, model_versions)
+
+    @freeze_time("2021-11-11 11:11:11.111000")
+    def test_list_model_version(self):
+        name = "test_list_MV"
+        self.store.create_registered_model(name)
+        tags = ["tag1", "tag2", "tag3"]
+        mvs = [
+            self.store.create_model_version(name, "path/to/source", "test", "application_1234"),
+            self.store.create_model_version(
+                name, "path/to/source", "test", "application_1234", tags=[tags[0]]
+            ),
+            self.store.create_model_version(
+                name, "path/to/source", "test", "application_1234", tags=[tags[1]]
+            ),
+            self.store.create_model_version(
+                name, "path/to/source", "test", "application_1234", tags=[tags[0], tags[2]]
+            ),
+            self.store.create_model_version(
+                name, "path/to/source", "test", "application_1234", tags=tags
+            ),
+        ]
+
+        results = self.store.list_model_version(name)
+        self.assertEqual(len(results), 5)
+        self._compare_model_versions(results, mvs)
+
+        results = self.store.list_model_version(name, filter_tags=tags[0:1])
+        self.assertEqual(len(results), 3)
+        self._compare_model_versions(results, [mvs[1], mvs[3], mvs[4]])
+
+        results = self.store.list_model_version(name, filter_tags=tags[0:2])
+        self.assertEqual(len(results), 1)
+        self._compare_model_versions(results, [mvs[-1]])
+
+        # empty result
+        other_tag = ["tag4"]
+        results = self.store.list_model_version(name, filter_tags=other_tag)
+        self.assertEqual(len(results), 0)
+        self.assertEqual(results, [])
+
+        # empty result
+        results = self.store.list_registered_model(filter_tags=tags + other_tag)
+        self.assertEqual(len(results), 0)
+        self.assertEqual(results, [])
```

Comment:
This test only contains one registered model name. Adding other registered model name helps to test list_model_version thoroughly when both name and tags are provided.